torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2528,14 +2528,26 @@ XLATensorPtr XLATensor::trace(const XLATensorPtr& input) {
 XLATensorPtr XLATensor::transpose(const XLATensorPtr& input, int64_t dim0,
                                   int64_t dim1) {
   auto input_shape = input->shape();
-  auto permute_dims = torch::lazy::MakeTransposePermutation(
-      /*dim0=*/dim0, /*dim1=*/dim1, /*rank=*/input_shape.get().rank());
-  ViewInfo view_info(ViewInfo::Type::kPermute, input_shape, permute_dims);
+  ViewInfo view_info;
+  if (input_shape.get().rank() <= 1) {
+    // return a view of self if input rank <=1
+    torch::lazy::Value ir_value = input->GetIrValue();
```

Comment:
Just curious what's the benefits of not doing it within `MakeTransposePermutation`?

Reply:
I am not sure if `kPermute` can handle two same dimension. We are in the middle of the functionization refactoring so I don't want to touch the view code too much hence this quick fix. We can revisit after refactoring is done. @wonjoolee95 fyi.